### PR TITLE
#1401 move env files to a separate folder

### LINKED ISSUE #1401: add `modin/env` folder for files with pip and conda packages
At the moment, we have several environment files that are located in the root of the repository. Example: `df_test_requires.txt`, `env_windows.yml`, `requirements.txt`, `windows_test_requires.txt`.

What do you think about adding a folder for these files? 

- This will reduce the semantic load for users who first see the project. 
- Also in the future there will be less chances to forget to add the package to any file.

### PR DESCRIPTION
**Pull request by dchigarev**
Signed-off-by: Dmitry Chigarev <dmitry.chigarev@intel.com>

## What do these changes do?

- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #1401 
- [ ] tests added and passing

### CODE REVIEW

File: ci/jenkins/build-tests/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 FROM python:3.6.6-stretch
 
-COPY requirements.txt requirements.txt
-RUN pip install -r requirements.txt
+COPY env/requirements.txt requirements.txt
```
Comment (anmyachev) [inline ci/jenkins/build-tests/Dockerfile:3]:
```suggestion
COPY env/requirements.txt env/requirements.txt
```

File: ci/jenkins/performance-tests/Dockerfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 FROM python:3.6.6-stretch
 
-COPY requirements.txt requirements.txt
-RUN pip install -r requirements.txt
+COPY env/requirements.txt requirements.txt
```
Comment (anmyachev) [inline ci/jenkins/performance-tests/Dockerfile:3]:
```suggestion
COPY env/requirements.txt env/requirements.txt
```

**Comment by dchigarev** [comment]
we probably should change path to `requirements.txt` at TeamCity internal build scripts, which seems to be are not represented in Modin repository

**Comment by anmyachev** [comment]
@devin-petersohn can we change the path to `requirements.txt` in TeamCity build scripts?

**Comment by devin-petersohn** [comment]
@anmyachev Yes, but we will have to block all other PRs on this change. I will help with this when we are ready to merge.

File: requirements/df_test_requires.txt [hunk 3]
```diff
@@ -1,13 +1,12 @@
...
-pytest
-pytest-testmon
-pytest-custom_exit_code
-pytest-xdist
-matplotlib
-xarray
-scipy
-Jinja2
-msgpack<1.0
-pandas_gbq
```
Comment (devin-petersohn) [inline requirements/df_test_requires.txt:13]: I think we want to keep this.

**Pull request merged**